### PR TITLE
refactor price.js

### PR DESCRIPTION
**Pull request by panleone**
## Abstract

A bit of refactor for the price class and its relation with settings:

- the circular dependency between settings.js and price.js has been removed, now the oracle does not know about the settings and viceversa.
-  call the function `fillCurrencySelect` only once
-  load the oracle on construction.
- make `fLoadedCurrencies` private

---

## Testing

Test that price is still working as before

### CODE REVIEW

File: scripts/prices.js [hunk 1]
```diff
@@ -105,33 +108,26 @@ export class Oracle {
...
             return this.getCachedCurrencies();
         }
     }
+
+    async load() {
+        while (!this.#fLoadedCurrencies) {
+            await this.getCurrencies();
+            if (!this.#fLoadedCurrencies) await sleep(5000);
+        }
+        getEventEmitter().emit('currency-loaded', this.mapCurrencies);
```
Comment (JSKitty) [inline scripts/prices.js:125]:
```suggestion
        // Update any listeners for the full currency list (Settings, etc)
        getEventEmitter().emit('currency-loaded', this.mapCurrencies);
        // Update the balance to render the price instantly
        getEventEmitter().emit('balance-update');
```

Small suggestion: even after `load()` is successful after multiple failed attempts, it takes another >15s before `balance-update` is called by `global.js`, adding a delay to the currency displaying in the Dashboard, we can render both the Balances and Settings here.

File: scripts/settings.js [hunk 2]
```diff
@@ -36,11 +35,7 @@ function getDefaultCurrency() {
     const langCode = navigator.languages[0]?.split('-')?.at(-1) || 'US';
     return getCurrencyByAlpha2(langCode)?.toLowerCase() || 'usd';
 }
-/**
- * The user-selected Price Oracle, used for all price data
- * @type {Oracle}
- */
-export let cOracle = new Oracle();
+
```
Comment (JSKitty) [inline scripts/settings.js:38]: If we're removing the Oracle instantiation from here, we should at least add some method or settings variable for us to "change" between instances in real-time via Settings in the future, as this is not intended to be a hardcoded single class, it should function similar to `cExplorer` or `cNode`, which are both stored in `settings.js`.

**Review by JSKitty — COMMENTED** [review]
Generally I like the decoupling of modules, and this implementation is good.

I am unsure of what we should do to allow Oracles to be chosen via the Settings though, because this implementation differs from how `cNetwork` and `cNode` work, when they're all functionally the same: different endpoints.

File: scripts/settings.js (hunk 2 above)
Comment (panleone) [reply scripts/settings.js:38]: Having a setting to choose where you want to get the price from maybe is a bit too much...
Regardless we can do this change when we will actually implement the feature?

Comment (JSKitty) [reply scripts/settings.js:38]: I think users should have full control over their data 🤷‍♂️ 

And sure, we can save it for when more are hosted.

**Review by JSKitty — APPROVED** [review]
**tACK** 8c2a8d76a2125918124f2617f35dc716230c60b2

**Pull request merged**